feat!: better simulateutility typings

Thunkar opened the pull request:
Closes: https://linear.app/aztec-labs/issue/F-74/simulateutility-typings, https://github.com/AztecProtocol/aztec-packages/issues/11275

Improves the typings for utilities, both in PXE and in the wallet interface. Makes sure only encoded data goes through the wire (mimicking how simulateTx works) and lets the "consumers" do the decoding.

Thunkar commented on yarn-project/aztec.js/src/utils/authwit.ts:
```diff
@@ -171,7 +171,18 @@ export async function lookupValidity(
     isInternal: false,
     isStatic: false,
-    parameters: [{ name: 'message_hash', type: { kind: 'field' }, visibility: 'private' as ABIParameterVisibility }],
+    parameters: [
```
This was a preexisting bug masked by PXE recomputing the selector from the actual ABI. yikes

benesjan commented on yarn-project/pxe/src/contract_function_simulator/contract_function_simulator.ts:
```diff
@@ -217,7 +217,7 @@ export class ContractFunctionSimulator {
    * accounts if not specified.
    * @returns A decoded ABI value containing the function's return data.
```
```suggestion
   * @returns A return value of the utility function in a form as returned by the simulator (Noir fields)
```

benesjan reviewed (approved):
Nice and reasonable

Thunkar committed:
feat!: better simulateutility typings

Closes: https://linear.app/aztec-labs/issue/F-74/simulateutility-typings, https://github.com/AztecProtocol/aztec-packages/issues/11275

Improves the typings for utilities, both in PXE and in the wallet interface. Makes sure only encoded data goes through the wire (mimicking how simulateTx works) and lets the "consumers" do the decoding.

Co-authored-by: Gregorio Juliana <gregojquiros@gmail.com>
Co-authored-by: thunkar <gregojquiros@gmail.com>
